addons/purchase/models/purchase.py

```diff
@@ -189,6 +194,23 @@ def _compute_date_planned(self):
             else:
                 order.date_planned = False
 
+    @api.depends('partner_id', 'company_id')
+    def _compute_fiscal_position_id(self):
+        """
+        Trigger the change of fiscal position when the address is modified.
+        """
+        cache = {}
+        for order in self:
+            if not order.partner_id:
+                order.fiscal_position_id = False
+                continue
+            key = (order.company_id.id, order.partner_id.id)
+            if key not in cache:
+                cache[key] = self.env['account.fiscal.position'].with_company(
+                    order.company_id
+                )._get_fiscal_position(order.partner_id)
+            order.fiscal_position_id = cache[key]
```

Comment:
I think it's a bit risky to change this in stable. There are already a lot of spots where the `fiscal_position_id` is being pre-calculated and passed as the create values for a PO (these are only a couple of them, there are more):

https://github.com/odoo/odoo/blob/9d4f8b2fda5c75438ad30a08b4f7d95eac4f2f23/addons/sale_purchase/models/sale_order_line.py#L129

https://github.com/odoo/odoo/blob/9d4f8b2fda5c75438ad30a08b4f7d95eac4f2f23/addons/purchase_stock/models/stock_rule.py#L284

Also, at this point there may also be lots of customizations that depend on the current onchange behavior/pre-calc create values.

Final also, `order.partner_id` is a mandatory field, so this compute is always going to go through every partner/company key combo.

Reply:
Never mind, I just took another look, and we don't even need that, I don't know why I did that in the first place. Thanks!